lib/src/Uncertainty/Algorithm/MetaModel/Kriging/openturns/GaussianProcessConditionalCovariance.hxx

```diff
@@ -0,0 +1,129 @@
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_GaussianProcessConditionalCovariance_HXX
+#define OPENTURNS_GaussianProcessConditionalCovariance_HXX
+
+#include "openturns/GaussianProcessRegressionResult.hxx"
+#include "openturns/CovarianceModel.hxx"
+#include "openturns/PersistentCollection.hxx"
+#include "openturns/Indices.hxx"
+#include "openturns/Function.hxx"
+#include "openturns/Normal.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class GaussianProcessConditionalCovariance
+ *
+ * The result of a chaos expansion
+ */
+
+class OT_API GaussianProcessConditionalCovariance
+  : public PersistentObject
+{
+  CLASSNAME
+
+public:
+
+  typedef GaussianProcessFitterResult::LinearAlgebra LinearAlgebra;
+  typedef Collection<CovarianceMatrix> CovarianceMatrixCollection;
+
+  /** Default constructor */
+  GaussianProcessConditionalCovariance();
+
+  /** Parameter constructor without any cholesky factor*/
+  GaussianProcessConditionalCovariance(const GaussianProcessRegressionResult & result);
+
+  /** Virtual constructor */
+  GaussianProcessConditionalCovariance * clone() const override;
+
+  /** String converter */
+  String __repr__() const override;
+  String __str__(const String & offset = "") const override;
+
+  /** Compute mean of new points conditionally to observations */
+  virtual Sample getConditionalMean(const Sample & xi) const;
+
+  /** Compute mean of new points conditionally to observations */
+  virtual Point getConditionalMean(const Point & xi) const;
+
+  /** Compute covariance matrix conditionally to observations*/
+  virtual CovarianceMatrix getConditionalCovariance(const Sample & xi) const ;
+
+  /** Compute covariance matrix conditionally to observations*/
+  virtual CovarianceMatrix getConditionalCovariance(const Point & xi) const;
+
+  /** Compute covariance matrices conditionally to observations (1 cov / point)*/
+  virtual CovarianceMatrixCollection getConditionalMarginalCovariance(const Sample & xi) const;
+
+  /** Compute covariance matrix conditionally to observations (1 cov of size outdimension)*/
+  virtual CovarianceMatrix getConditionalMarginalCovariance(const Point & xi) const;
+
+  /** Compute marginal variance conditionally to observations (1 cov of size outdimension)*/
+  virtual Scalar getConditionalMarginalVariance(const Point & point,
```

Comment:
This method is currently uncovered by the current unit tests (discovered by reading the output from CodeCov).